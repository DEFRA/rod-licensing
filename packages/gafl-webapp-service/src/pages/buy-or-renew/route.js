import { BUY_OR_RENEW } from '../../uri.js'
import Joi from 'joi'
import pageRoute from '../../routes/page-route.js'
import { nextPage } from '../../routes/next-page.js'
export const buyOrRenew = {
  buy: 'buy',
  renew: 'renew'
}

export const validator = Joi.object({
  'buy-or-renew': Joi.string()
    .valid('buy-licence', 'renew-licence')
    .required()
}).options({ abortEarly: false, allowUnknown: true })

const route = pageRoute(BUY_OR_RENEW.page, BUY_OR_RENEW.uri, validator, nextPage)

export default route
