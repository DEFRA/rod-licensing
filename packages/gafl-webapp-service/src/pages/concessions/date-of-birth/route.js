import { DATE_OF_BIRTH } from '../../../uri.js'
import Joi from 'joi'
import pageRoute from '../../../routes/page-route.js'
import { validation } from '@defra-fish/business-rules-lib'
import { nextPage } from '../../../routes/next-page.js'

export const validator = payload => {
  const dateOfBirth = `${payload['date-of-birth-year']}-${payload['date-of-birth-month']}-${payload['date-of-birth-day']}`
  Joi.assert(
    { 'date-of-birth': dateOfBirth },
    Joi.object({
      'date-of-birth': validation.contact.createBirthDateValidator(Joi)
    })
  )
}

export const getData = async request => {
  const { isLicenceForYou } = await request.cache().helpers.status.getCurrentPermission()

  return { isLicenceForYou }
}

export default pageRoute(DATE_OF_BIRTH.page, DATE_OF_BIRTH.uri, validator, nextPage, getData)
