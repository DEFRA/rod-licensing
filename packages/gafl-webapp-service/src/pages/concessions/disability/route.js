import { DISABILITY_CONCESSION } from '../../../uri.js'
import pageRoute from '../../../routes/page-route.js'
import Joi from 'joi'
import { validation } from '@defra-fish/business-rules-lib'
import * as concessionHelper from '../../../processors/concession-helper.js'
import { nextPage } from '../../../routes/next-page.js'
import { disabilityConcessionTypes } from './update-transaction.js'

const validator = Joi.object({
  'disability-concession': Joi.string()
    .valid(...Object.values(disabilityConcessionTypes))
    .required(),
  'ni-number': Joi.alternatives().conditional('disability-concession', {
    is: disabilityConcessionTypes.pipDla,
    then: validation.contact.createNationalInsuranceNumberValidator(Joi).required(),
    otherwise: Joi.string().empty('')
  }),
  'blue-badge-number': Joi.alternatives().conditional('disability-concession', {
    is: disabilityConcessionTypes.blueBadge,
    then: Joi.string()
      .max(25)
      .required(),
    otherwise: Joi.string().empty('')
  })
}).options({ abortEarly: false, allowUnknown: true })

export const getData = async request => {
  const permission = await request.cache().helpers.transaction.getCurrentPermission()

  return {
    hasJunior: concessionHelper.hasJunior(permission),
    hasSenior: concessionHelper.hasSenior(permission),
    isLicenceForYou: permission.isLicenceForYou,
    ...disabilityConcessionTypes
  }
}

export default pageRoute(DISABILITY_CONCESSION.page, DISABILITY_CONCESSION.uri, validator, nextPage, getData)
