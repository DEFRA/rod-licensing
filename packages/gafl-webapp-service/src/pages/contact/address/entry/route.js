import { ADDRESS_ENTRY, ADDRESS_LOOKUP } from '../../../../uri.js'
import pageRoute from '../../../../routes/page-route.js'
import Joi from 'joi'
import { validation } from '@defra-fish/business-rules-lib'
import { countries } from '../../../../processors/refdata-helper.js'
import { nextPage } from '../../../../routes/next-page.js'
import { getPronoun } from '../../../../processors/licence-type-display.js'

export const validator = Joi.object({
  premises: validation.contact.createPremisesValidator(Joi),
  street: validation.contact.createStreetValidator(Joi),
  locality: validation.contact.createLocalityValidator(Joi),
  town: validation.contact.createTownValidator(Joi),
  postcode: Joi.alternatives().conditional('country-code', {
    is: Joi.string().pattern(/^GB/),
    then: validation.contact.createUKPostcodeValidator(Joi),
    otherwise: validation.contact.createOverseasPostcodeValidator(Joi)
  }),
  'country-code': Joi.string().required()
}).options({ abortEarly: false, allowUnknown: true })

export const getCountryDropDownOptions = async () => {
  const options = await countries.getAll()
  return options.filter(country => country.code !== 'GB')
}

export const getData = async request => {
  const { addresses, searchTerms } = await request.cache().helpers.addressLookup.getCurrentPermission()
  const { isLicenceForYou } = await request.cache().helpers.transaction.getCurrentPermission()

  const pronoun = getPronoun(isLicenceForYou).possessive

  return {
    pronoun,
    searchTerms: !addresses?.length && searchTerms ? searchTerms : null,
    countries: await getCountryDropDownOptions(),
    lookupPage: ADDRESS_LOOKUP.uri
  }
}

export default pageRoute(ADDRESS_ENTRY.page, ADDRESS_ENTRY.uri, validator, nextPage, getData)
