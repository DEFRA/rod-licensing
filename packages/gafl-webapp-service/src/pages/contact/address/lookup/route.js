import { ADDRESS_LOOKUP, ADDRESS_ENTRY, OS_TERMS } from '../../../../uri.js'
import pageRoute from '../../../../routes/page-route.js'
import Joi from 'joi'
import { hasJunior, validation } from '@defra-fish/business-rules-lib'
import { isPhysical } from '../../../../processors/licence-type-display.js'
import { nextPage } from '../../../../routes/next-page.js'
import { addLanguageCodeToUri } from '../../../../processors/uri-helper.js'

const validator = Joi.object({
  premises: validation.contact.createPremisesValidator(Joi),
  postcode: validation.contact.createUKPostcodeValidator(Joi)
}).options({ abortEarly: false, allowUnknown: true })

export const getData = async request => {
  const permission = await request.cache().helpers.transaction.getCurrentPermission()

  return {
    isLicenceForYou: permission.isLicenceForYou,
    licenceLength: permission.licenceLength,
    junior: hasJunior(permission),
    isPhysical: isPhysical(permission),
    uri: {
      entryPage: addLanguageCodeToUri(request, ADDRESS_ENTRY.uri),
      osTerms: addLanguageCodeToUri(request, OS_TERMS.uri)
    }
  }
}

export default pageRoute(ADDRESS_LOOKUP.page, ADDRESS_LOOKUP.uri, validator, nextPage, getData)
