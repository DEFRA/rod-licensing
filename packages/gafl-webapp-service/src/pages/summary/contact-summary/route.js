import pageRoute from '../../../routes/page-route.js'
import GetDataRedirect from '../../../handlers/get-data-redirect.js'
import { countries } from '../../../processors/refdata-helper.js'
import { HOW_CONTACTED } from '../../../processors/mapping-constants.js'
import { CONTACT_SUMMARY_SEEN } from '../../../constants.js'
import { isPhysical } from '../../../processors/licence-type-display.js'
import { nextPage } from '../../../routes/next-page.js'

import {
  CONTACT_SUMMARY,
  LICENCE_SUMMARY,
  NAME,
  ADDRESS_ENTRY,
  ADDRESS_SELECT,
  ADDRESS_LOOKUP,
  CONTACT,
  NEWSLETTER,
  LICENCE_FULFILMENT,
  LICENCE_CONFIRMATION_METHOD
} from '../../../uri.js'

const getData = async request => {
  const status = await request.cache().helpers.status.getCurrentPermission()
  const permission = await request.cache().helpers.transaction.getCurrentPermission()

  if (!status.renewal) {
    if (!status[NAME.page]) {
      throw new GetDataRedirect(NAME.uri)
    }

    if (!status[ADDRESS_ENTRY.page] && !status[ADDRESS_SELECT.page]) {
      throw new GetDataRedirect(ADDRESS_LOOKUP.uri)
    }

    if (!status[CONTACT.page]) {
      throw new GetDataRedirect(CONTACT.uri)
    }

    if (!status[NEWSLETTER.page]) {
      throw new GetDataRedirect(NEWSLETTER.uri)
    }
  }

  status.fromSummary = CONTACT_SUMMARY_SEEN
  await request.cache().helpers.status.setCurrentPermission(status)
  const countryName = await countries.nameFromCode(permission.licensee.countryCode)

  return {
    summaryTable: getLicenseeDetailsSummaryRows(permission, countryName),
    uri: {
      licenceSummary: LICENCE_SUMMARY.uri
    }
  }
}

export default pageRoute(CONTACT_SUMMARY.page, CONTACT_SUMMARY.uri, null, nextPage, getData)

export const getLicenseeDetailsSummaryRows = (permission, countryName) => {
  return [
    getRow('Name', `${permission.licensee.firstName} ${permission.licensee.lastName}`, NAME.uri, 'name', 'change-name'),
    getRow('Address', getAddressText(permission.licensee, countryName), ADDRESS_LOOKUP.uri, 'address', 'change-address'),
    ...getContactDetails(permission),
    getRow(
      'Newsletter',
      permission.licensee.preferredMethodOfNewsletter !== HOW_CONTACTED.none ? 'Yes' : 'No',
      NEWSLETTER.uri,
      'newsletter',
      'change-newsletter'
    )
  ]
}

const CONTACT_TEXT_DEFAULT = {
  EMAIL: 'Email to ',
  TEXT: 'Text message to ',
  DEFAULT: 'Note of licence'
}

const CONTACT_TEXT_NON_PHYSICAL = {
  EMAIL: CONTACT_TEXT_DEFAULT.EMAIL,
  TEXT: 'Text messages to ',
  DEFAULT: 'Make a note on confirmation'
}

const CONTACT_TEXT_PHYSICAL = {
  EMAIL: CONTACT_TEXT_DEFAULT.EMAIL,
  TEXT: 'Text messages to ',
  DEFAULT: 'By post'
}

const CHANGE_CONTACT = 'change-contact'

const getContactDetails = permission => {
  if (isPhysical(permission)) {
    if (permission.licensee.postalFulfilment) {
      return [
        getRow('Licence', 'By post', LICENCE_FULFILMENT.uri, 'licence fulfilment option', 'change-licence-fulfilment-option'),
        getRow(
          'Licence Confirmation',
          getContactText(permission.licensee.preferredMethodOfConfirmation, permission.licensee),
          LICENCE_CONFIRMATION_METHOD.uri,
          'licence confirmation option',
          'change-licence-confirmation-option'
        ),
        getRow(
          'Contact',
          getContactText(permission.licensee.preferredMethodOfReminder, permission.licensee, CONTACT_TEXT_PHYSICAL),
          CONTACT.uri,
          'contact',
          CHANGE_CONTACT
        )
      ]
    } else {
      return [
        getRow(
          'Licence',
          getContactText(permission.licensee.preferredMethodOfConfirmation, permission.licensee),
          LICENCE_CONFIRMATION_METHOD.uri,
          'licence confirmation option',
          'change-licence-confirmation-option'
        ),
        getRow(
          'Contact',
          getContactText(permission.licensee.preferredMethodOfReminder, permission.licensee, CONTACT_TEXT_PHYSICAL),
          CONTACT.uri,
          'contact',
          CHANGE_CONTACT
        )
      ]
    }
  } else {
    return [
      getRow(
        'Licence details',
        getContactText(permission.licensee.preferredMethodOfReminder, permission.licensee, CONTACT_TEXT_NON_PHYSICAL),
        CONTACT.uri,
        'contact',
        CHANGE_CONTACT
      )
    ]
  }
}

const getAddressText = (licensee, countryName) =>
  [licensee.premises, licensee.street, licensee.locality, licensee.town, licensee.postcode, countryName?.toUpperCase()]
    .filter(Boolean)
    .join(', ')

const getContactText = (contactMethod, licensee, contactText = CONTACT_TEXT_DEFAULT) => {
  switch (contactMethod) {
    case HOW_CONTACTED.email:
      return contactText.EMAIL + licensee.email
    case HOW_CONTACTED.text:
      return contactText.TEXT + licensee.mobilePhone
    default:
      return contactText.DEFAULT
  }
}

const getRow = (label, text, href, visuallyHiddenText, id) => {
  return {
    key: {
      text: label
    },
    value: {
      text
    },
    ...(href && {
      actions: {
        items: [
          {
            href,
            text: 'Change',
            visuallyHiddenText: visuallyHiddenText,
            attributes: { id }
          }
        ]
      }
    })
  }
}
