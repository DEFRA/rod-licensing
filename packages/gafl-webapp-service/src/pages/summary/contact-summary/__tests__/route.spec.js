import { getLicenseeDetailsSummaryRows } from '../route'

const address = {
  firstName: 'Fester',
  lastName: 'Tester',
  premises: '14 Howecroft Court',
  street: 'Eastmead Lane',
  town: 'Bristol',
  postcode: 'BS9 1HJ'
}

describe('contact-summary > route', () => {
  describe('getLicenseeDetailsSummaryRows', () => {
    describe('when purchasing a 12 month (physical licence) with postal fulfilment', () => {
      it('should display the Licence as post, Licence Confirmation and Contact as the email and Newsletter as no', () => {
        const permission = {
          licenceLength: '12M',
          licensee: {
            ...address,
            postalFulfilment: true,
            preferredMethodOfConfirmation: 'Email',
            preferredMethodOfReminder: 'Email',
            email: 'new3@example.com',
            preferredMethodOfNewsletter: 'Prefer not to be contacted'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })

      it('should display the Licence as post, Licence Confirmation and Contact as the phone number and Newsletter as yes', () => {
        const permission = {
          licenceLength: '12M',
          licensee: {
            ...address,
            postalFulfilment: true,
            preferredMethodOfConfirmation: 'Text',
            preferredMethodOfReminder: 'Text',
            mobilePhone: '07700900900',
            preferredMethodOfNewsletter: 'Yes'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })

      it('should display the Licence as post, Licence Confirmation as note of licence and Contact as post', () => {
        const permission = {
          licenceLength: '12M',
          licensee: {
            ...address,
            postalFulfilment: true,
            preferredMethodOfConfirmation: 'Prefer not to be contacted',
            preferredMethodOfReminder: 'Letter'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })
    })

    describe('when purchasing a 12 month (physical licence) without postal fulfilment', () => {
      it('should display the Licence and Contact as the email and Newsletter as no', () => {
        const permission = {
          licenceLength: '12M',
          licensee: {
            ...address,
            postalFulfilment: false,
            preferredMethodOfConfirmation: 'Email',
            preferredMethodOfReminder: 'Email',
            email: 'new3@example.com',
            preferredMethodOfNewsletter: 'Prefer not to be contacted'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })

      it('should display the Licence and Contact as the phone number and Newsletter as yes', () => {
        const permission = {
          licenceLength: '12M',
          licensee: {
            ...address,
            postalFulfilment: false,
            preferredMethodOfConfirmation: 'Text',
            preferredMethodOfReminder: 'Text',
            mobilePhone: '07700900900',
            preferredMethodOfNewsletter: 'Yes'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })
    })

    describe('when purchasing a 1 day (non physical licence)', () => {
      it('should display the Contact as the email and Newsletter as no', () => {
        const permission = {
          licenceLength: '1D',
          licensee: {
            ...address,
            preferredMethodOfConfirmation: 'Email',
            preferredMethodOfReminder: 'Email',
            email: 'new3@example.com',
            preferredMethodOfNewsletter: 'Prefer not to be contacted'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })

      it('should display the Contact as the phone number and Newsletter as yes', () => {
        const permission = {
          licenceLength: '1D',
          licensee: {
            ...address,
            postalFulfilment: false,
            preferredMethodOfConfirmation: 'Text',
            preferredMethodOfReminder: 'Text',
            mobilePhone: '07700900900',
            preferredMethodOfNewsletter: 'Yes'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })

      it('should display the Contact as Make a note on confirmation', () => {
        const permission = {
          licenceLength: '1D',
          licensee: {
            ...address,
            postalFulfilment: false,
            preferredMethodOfConfirmation: 'Prefer not to be contacted',
            preferredMethodOfReminder: 'Prefer not to be contacted',
            preferredMethodOfNewsletter: 'Yes'
          }
        }
        const summaryTable = getLicenseeDetailsSummaryRows(permission, 'GB')
        expect(summaryTable).toMatchSnapshot()
      })
    })
  })
})
