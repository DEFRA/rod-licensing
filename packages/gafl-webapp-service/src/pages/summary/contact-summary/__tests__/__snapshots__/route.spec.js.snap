// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 1 day (non physical licence) should display the Contact as Make a note on confirmation 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Licence details",
    },
    "value": Object {
      "text": "Make a note on confirmation",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "Yes",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 1 day (non physical licence) should display the Contact as the email and Newsletter as no 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Licence details",
    },
    "value": Object {
      "text": "Email to new3@example.com",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "No",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 1 day (non physical licence) should display the Contact as the phone number and Newsletter as yes 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Licence details",
    },
    "value": Object {
      "text": "Text messages to 07700900900",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "Yes",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 12 month (physical licence) with postal fulfilment should display the Licence as post, Licence Confirmation and Contact as the email and Newsletter as no 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-fulfilment-option",
          },
          "href": "/buy/fulfilment",
          "text": "Change",
          "visuallyHiddenText": "licence fulfilment option",
        },
      ],
    },
    "key": Object {
      "text": "Licence",
    },
    "value": Object {
      "text": "By post",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-confirmation-option",
          },
          "href": "/buy/confirmation-method",
          "text": "Change",
          "visuallyHiddenText": "licence confirmation option",
        },
      ],
    },
    "key": Object {
      "text": "Licence Confirmation",
    },
    "value": Object {
      "text": "Email to new3@example.com",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Contact",
    },
    "value": Object {
      "text": "Email to new3@example.com",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "No",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 12 month (physical licence) with postal fulfilment should display the Licence as post, Licence Confirmation and Contact as the phone number and Newsletter as yes 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-fulfilment-option",
          },
          "href": "/buy/fulfilment",
          "text": "Change",
          "visuallyHiddenText": "licence fulfilment option",
        },
      ],
    },
    "key": Object {
      "text": "Licence",
    },
    "value": Object {
      "text": "By post",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-confirmation-option",
          },
          "href": "/buy/confirmation-method",
          "text": "Change",
          "visuallyHiddenText": "licence confirmation option",
        },
      ],
    },
    "key": Object {
      "text": "Licence Confirmation",
    },
    "value": Object {
      "text": "Text message to 07700900900",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Contact",
    },
    "value": Object {
      "text": "Text messages to 07700900900",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "Yes",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 12 month (physical licence) with postal fulfilment should display the Licence as post, Licence Confirmation as note of licence and Contact as post 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-fulfilment-option",
          },
          "href": "/buy/fulfilment",
          "text": "Change",
          "visuallyHiddenText": "licence fulfilment option",
        },
      ],
    },
    "key": Object {
      "text": "Licence",
    },
    "value": Object {
      "text": "By post",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-confirmation-option",
          },
          "href": "/buy/confirmation-method",
          "text": "Change",
          "visuallyHiddenText": "licence confirmation option",
        },
      ],
    },
    "key": Object {
      "text": "Licence Confirmation",
    },
    "value": Object {
      "text": "Note of licence",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Contact",
    },
    "value": Object {
      "text": "By post",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "Yes",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 12 month (physical licence) without postal fulfilment should display the Licence and Contact as the email and Newsletter as no 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-confirmation-option",
          },
          "href": "/buy/confirmation-method",
          "text": "Change",
          "visuallyHiddenText": "licence confirmation option",
        },
      ],
    },
    "key": Object {
      "text": "Licence",
    },
    "value": Object {
      "text": "Email to new3@example.com",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Contact",
    },
    "value": Object {
      "text": "Email to new3@example.com",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "No",
    },
  },
]
`;

exports[`contact-summary > route getLicenseeDetailsSummaryRows when purchasing a 12 month (physical licence) without postal fulfilment should display the Licence and Contact as the phone number and Newsletter as yes 1`] = `
Array [
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-name",
          },
          "href": "/buy/name",
          "text": "Change",
          "visuallyHiddenText": "name",
        },
      ],
    },
    "key": Object {
      "text": "Name",
    },
    "value": Object {
      "text": "Fester Tester",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-address",
          },
          "href": "/buy/find-address",
          "text": "Change",
          "visuallyHiddenText": "address",
        },
      ],
    },
    "key": Object {
      "text": "Address",
    },
    "value": Object {
      "text": "14 Howecroft Court, Eastmead Lane, Bristol, BS9 1HJ, GB",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-licence-confirmation-option",
          },
          "href": "/buy/confirmation-method",
          "text": "Change",
          "visuallyHiddenText": "licence confirmation option",
        },
      ],
    },
    "key": Object {
      "text": "Licence",
    },
    "value": Object {
      "text": "Text message to 07700900900",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-contact",
          },
          "href": "/buy/contact",
          "text": "Change",
          "visuallyHiddenText": "contact",
        },
      ],
    },
    "key": Object {
      "text": "Contact",
    },
    "value": Object {
      "text": "Text messages to 07700900900",
    },
  },
  Object {
    "actions": Object {
      "items": Array [
        Object {
          "attributes": Object {
            "id": "change-newsletter",
          },
          "href": "/buy/newsletter",
          "text": "Change",
          "visuallyHiddenText": "newsletter",
        },
      ],
    },
    "key": Object {
      "text": "Newsletter",
    },
    "value": Object {
      "text": "Yes",
    },
  },
]
`;
